R/data-objects.R

```diff
@@ -152,48 +308,32 @@ iget <- function(
     # chunk size
     chunk_size <- calc_chunk_size(file_size, count)
     # vectorised call
-    resp <- Map(function(offset, count) {
-      iget_(
-        logical_path = logical_path,
-        offset = offset,
-        count = count,
-        verbose = verbose
-      )
-    },
-    chunk_size[[2]],
-    chunk_size[[3]])
-    # fuse files
-    resp <- fuse_file(resp)
-  } else {
-    resp <-
-      iget_(
-        logical_path = logical_path,
-        offset = offset,
-        count = count,
-        verbose = verbose
-      )
-  }
+    resp <- Map(
+      function(offset, count) {
+        irods_to_local_(
+          logical_path = logical_path,
+          offset = offset,
+          count = count,
+          verbose = verbose
+      )},
+      chunk_size[[2]],
+      chunk_size[[3]]
+    )
 
-  # convert to file or R object
-  if (is.character(logical_path) && tools::file_ext(logical_path) != "rds") {
-    # check for local file/ R object
-    stop_local_overwrite(overwrite, path)
-    # close connection
-    on.exit(close(con))
-    # write file
-    con <- file(path, "wb")
-    writeBin(resp, con)
-    invisible(NULL)
+    # fuse files
+    fuse_file(resp)
   } else {
-    # close connection
-    on.exit(close(con))
-    # create R object
-    con <- rawConnection(resp)
-    readRDS(gzcon(con))
+    irods_to_local_(
```

Comment:
If the same strategy is used as in `local_to_irods()`, this can be merged with `local_to_irods_()`.

Reply:
Same as the previous comment. It separates the vectorised and non-vectorised portions, so I rather leave it.